### PR TITLE
added several new functions to nwnx_areas. added new utility function 'hexdump'.

### PR DESCRIPTION
**Pull request by stubhead**
added hexdump utility function to global library.
details:
- include/structs/CNWSArea.h - identified several unidentified fields
- plugins/areas/nwnx_areas.nss - added new functions: GetAreaTileset, GetAreaHeight, GetAreaWidth, GetAreaLighting, GetNoRest, Set/GetAreaListenMod, Set/GetAreaSpotMod, DumpArea
- plugins/areas/HookFunc.cpp - added new functions to get/set area data
- plugins/areas/HookFunc.h - modified to support above
- plugins/areas/NWNXAreas.cpp - modified to support above
- plugins/areas/NWNStructures.h - identified several unidentified fields in area struct
- plugins/funcs/funcs/area/f_GetFirstNextArea.c - fastidious tweak for global area counter
- plugins/funcs/funcs/other/f_DumpObject.c - updated to output info header for recognized object types
- lib/hexdump.c - new hexdump utility
- include/hexdump.h - new header file required to use new hexdump utility
- CMakeLists.txt - added hexdump to nwnx2.so

### CODE REVIEW

**Comment by stubhead** [comment]
i see that the software on the branch i forked from already had changes by niv that i hadn't noticed. i will rebase my fork, update the changes, and submit a new change request. please disregard this one.

**Comment by virusman** [comment]
Just force push the rebased commits to this branch, it will automaticallu update this PR.

**Comment by stubhead** [comment]
there are several conflicts to resolve first. looks like niv modified nwnx_areas using a radically different approach, where i based my additions off of the code you'd written. no worries, i should have my head around it soon.

**Comment by Th0mYT** [comment]
Force push 😰😰😰😰😰

What Have i just read?!?!?!? 

😂😂😂

Inviato da iPhone

**Comment by stubhead** [comment]
okay, i've run into a major problem. unfortunately, the new object-oriented approach for nwnx_areas provides an area class that's missing half the area structure [seriously], so there's no longer any way to get at the data i was successfully retrieving before. in addition, adding the data back into the class and adding assertions similar to those already present in order to validate the alignment of the data at compile time results in an error. finally, although there are methods for the new CNWSArea class that appear to duplicate the functions i've written, the methods appear to be only non-functional stubs. [indeed, w/half the data unavailable, they can't possibly work.]

to sum up, the new approach seems to be incompatible w/being able to get at the area data and i have neither the time nor the knowledge required to massage all my changes back in, in the new format.

happy to provide what i've done so far to whomever's been modifying this plugin in parallel w/me, in hopes it may be useful.

thoughts ?  ideas ?

**Comment by virusman** [comment]
Closing since the PR has been re-submitted at #128
